x-pack/plugins/ingest_pipelines/public/application/components/pipeline_processors_editor/components/load_from_json/modal_provider.tsx

```
@@ -0,0 +1,143 @@
+  },
+  error: {
+    title: i18n.translate('xpack.ingestPipelines.pipelineEditor.loadFromJson.error.title', {
+      defaultMessage: 'Invalid pipeline',
+    }),
+    body: i18n.translate('xpack.ingestPipelines.pipelineEditor.loadFromJson.error.body', {
+      defaultMessage: 'Please ensure the JSON is a valid pipeline object.',
+    }),
+  },
+};
+
+const defaultValue = {};
+const defaultValueRaw = JSON.stringify(defaultValue, null, 2);
+
+export const ModalProvider: FunctionComponent<Props> = ({ onDone, children }) => {
+  const [isModalVisible, setIsModalVisible] = useState(false);
+  const [isValidJson, setIsValidJson] = useState(true);
+  const [error, setError] = useState<Error | undefined>();
+  const jsonContent = useRef<Parameters<OnJsonEditorUpdateHandler>['0']>({
+    isValid: true,
+    validate: () => true,
+    data: {
+      format: () => defaultValue,
+      raw: defaultValueRaw,
+    },
+  });
+  const onJsonUpdate: OnJsonEditorUpdateHandler = (jsonUpdateData) => {
+    setIsValidJson(jsonUpdateData.validate());
+    jsonContent.current = jsonUpdateData;
+  };
+  return (
+    <>
+      {children(() => setIsModalVisible(true))}
+      {isModalVisible ? (
+        <EuiOverlayMask>
+          <EuiConfirmModal
+            data-test-subj="loadJsonConfirmationModal"
+            title={i18nTexts.modalTitle}
+            onCancel={() => {
+              setIsModalVisible(false);
+            }}
+            onConfirm={async () => {
+              try {
+                const json = jsonContent.current.data.format();
+                const { processors, on_failure: onFailure } = json;
+                // This function will throw if it cannot parse the pipeline object
+                deserialize({ processors, onFailure });
+                onDone(json as any);
+                setIsModalVisible(false);
+              } catch (e) {
+                setError(e);
+                // eslint-disable-next-line no-console
+                console.error(e);
```

Do we really need this console output? 😊

---

Thought it might be useful if users want to report an error, but I suppose it would be better to show this in a modal perhaps? Will remove.